pkg/values/value.go

```diff
@@ -38,9 +40,99 @@ func Wrap(v any) (Value, error) {
 		return NewInt64(tv), nil
 	case int:
 		return NewInt64(int64(tv)), nil
+	case big.Int:
+		return NewBigInt(tv), nil
 	case nil:
 		return nil, nil
 
+	// Better complex type support for maps
+	case map[string]string:
+		nm := map[string]any{}
+		for k, v := range tv {
+			nm[k] = v
+		}
+		return NewMap(nm)
+	case map[string]int:
+		nm := map[string]any{}
+		for k, v := range tv {
+			nm[k] = v
+		}
+		return NewMap(nm)
+	case map[string]int64:
+		nm := map[string]any{}
+		for k, v := range tv {
+			nm[k] = v
+		}
+		return NewMap(nm)
+	case map[string][]byte:
+		nm := map[string]any{}
+		for k, v := range tv {
+			nm[k] = v
+		}
+		return NewMap(nm)
+	case map[string]decimal.Decimal:
+		nm := map[string]any{}
+		for k, v := range tv {
+			nm[k] = v
+		}
+		return NewMap(nm)
+	case map[string]bool:
+		nm := map[string]any{}
+		for k, v := range tv {
+			nm[k] = v
+		}
+		return NewMap(nm)
+	case map[string]big.Int:
+		nm := map[string]any{}
+		for k, v := range tv {
+			nm[k] = v
+		}
+		return NewMap(nm)
+
+	// Better complex type support for lists
```

Comment:
Could this be made more efficient using reflection and recursion? I wanted to explore this but didn't get to it.

Reply:
Good point :) I reworked those sections, please take a look